straxen/plugins/nveto_hitlets.py

```diff
@@ -0,0 +1,143 @@
+    strax.Option(
+        'to_pe_file_nv',
+        default='/dali/lgrandi/wenz/strax_data/HdMdata_strax_v0_9_0/swt_gains.npy',  # noqa
+        help='URL of the to_pe conversion factors. Expect gains in units ADC/sample.'),
+)
+class nVETOHitlets(strax.Plugin):
+    """
+    Plugin which computes the nveto hitlets and their parameters.
+    """
+    __version__ = '0.0.2'
+
+    parallel = 'process'
+    rechunk_on_save = False
+    compressor = 'lz4'
+
+    depends_on = 'records_nv'
+
+    provides = 'hitlets_nv'
+    data_kind = 'hitlets_nv'
+
+    dtype = strax.hitlet_dtype()
+
+    def setup(self):
+        # TODO: Unify with TPC and add adc thresholds
+        self.to_pe = straxen.get_resource(self.config['to_pe_file_nv'], fmt='npy')
+
+    def compute(self, records_nv):
+        # Search again for hits in records:
+        hits = strax.find_hits(records_nv, min_amplitude=self.config['hit_min_amplitude_nv'])
+
+        # Merge concatenate overlapping  within a channel. This is important
+        # in case hits were split by record boundaries. In case we
+        # accidentally concatenate two PMT signals we split them later again.
+        hits = strax.concat_overlapping_hits(hits,
+                                             self.config['save_outside_hits_nv'],
+                                             self.config['channel_map']['nveto'])
+        hits = strax.sort_by_time(hits)
+
+        # Now convert hits into temp_hitlets including the data field:
+        if len(hits):
+            nsamples = hits['length'].max()
+        else:
+            nsamples = 0
+        temp_hitlets = np.zeros(len(hits), strax.hitlet_with_data_dtype(n_sample=nsamples))
+
+        strax.refresh_hit_to_hitlets(hits, temp_hitlets)
+        del hits
+
+        # Get hitlet data and split hitlets:
+        strax.get_hitlets_data(temp_hitlets, records_nv, to_pe=self.to_pe)
+
+        temp_hitlets = strax.split_peaks(temp_hitlets,
+                                         records_nv,
+                                         self.to_pe,
+                                         data_type='hitlets',
+                                         algorithm='local_minimum',
+                                         min_height=self.config['min_split_nv'],
+                                         min_ratio=self.config['min_split_ratio_nv']
+                                         )
+
+        # Compute other hitlet properties:
+        # We have to loop here 3 times over all hitlets...
+        strax.hitlet_properties(temp_hitlets)
+        entropy = strax.conditional_entropy(temp_hitlets, template='flat', square_data=False)
+        temp_hitlets['entropy'][:] = entropy
+        strax.compute_widths(temp_hitlets)
+
+        # Remove data field:
+        hitlets = np.zeros(len(temp_hitlets), dtype=strax.hitlet_dtype())
+        drop_data_field(temp_hitlets, hitlets)
+
+        return hitlets
+
+@numba.njit
+def drop_data_field(old_hitlets, new_hitlets):
```

Comment:
this one is very similar to refresh_hit_to_hitlets. Can we not make a function like
```python
for i in range(n):
    o = old_hitlets[i]
    n = new_hitlets[i]
    for name in o.dtype.names:
        if name != 'data': 
            if np.iterable(o['name']):
                n['name'][:] = o['name'][:] 
            else:
                n['name'] = o['name']
```

perhaps you tried already something like this but it feels a little awkward that you have to know all the dtypes by hart

Reply:
I would love if something like this works. But unfortunately you can call array.dtype.names in a numba function... But, let me check maybe something changed between numba 0.4.8 and 0.5

Reply:
See my comment in strax